### PR TITLE
 nixos/users:added users.allowLoginless option 

### LINKED ISSUE #73003: Not allowed to have a loginless system
**Describe the bug**
There are reasons to want a system, that has no user that can log in by SSH or with their password, for example in a declarative container

**To Reproduce**
1. set up a minimal declarative container
2. set `users.mutableUsers = false;`
3. nix asserts that I will be locked out from the system and refuses to build the configuration with
```
building Nix...
building the system configuration...
error:
Failed assertions:
- Neither the root account nor any wheel user has a password or SSH authorized key.
You must set one to prevent being locked out of your system.
```

**Expected behavior**
nix generates me a container that I can log in with `nixos-container root-login $containername`, or maybe I don't want to log into it at all (because it's a disposable VM or something of the like). I think there should be a check whether the generated system is a container and an additional "I know what I'm doing" override for other cases. I'll create a PR once we agreed on what's best.

Maintainer information:
```
yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: containers
```

Code causing the problem: https://github.com/NixOS/nixpkgs/blob/6445a7cf7d471e8629011b53d8a4a6999e68f7cc/nixos/modules/config/users-groups.nix#L576

mentioning @edolstra since you added the check.

### PR DESCRIPTION
**Pull request by pasqui23**
###### Motivation for this change
Fixes https://github.com/NixOS/nixpkgs/issues/73003
Also made security.initialRootPassword a renamed option

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: nixos/modules/rename.nix [hunk 1]
```diff
@@ -72,6 +72,7 @@ with lib;
     (mkRemovedOptionModule [ "security" "setuidPrograms" ] "Use security.wrappers instead")
 
     (mkRenamedOptionModule [ "security" "virtualization" "flushL1DataCache" ] [ "security" "virtualisation" "flushL1DataCache" ])
+    (mkRenamedOptionModule ["security" "initialRootPassword"] ["users" "users" "root" "initialHashedPassword"])
```
Comment (flokli) [inline nixos/modules/rename.nix]: please add this to `nixos/modules/config/users-groups.nix` instead, like done in other places - we intend to move away from a global `rename.nix`.

File: pkgs/tools/package-management/home-manager/default.nix [hunk 2]
```diff
@@ -6,13 +6,13 @@
 stdenv.mkDerivation rec {
 
   pname = "home-manager";
-  version = "2019-10-23";
+  version = "2019-10-29";
```
Comment (flokli) [inline pkgs/tools/package-management/home-manager/default.nix]: this probably isn't related, and should go in a separate PR ;-)

**Review by flokli — CHANGES_REQUESTED** [review]

**Comment by pasqui23** [comment]
Closed in favor of https://github.com/NixOS/nixpkgs/pull/73106
